core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -678,4 +716,13 @@ private[spark] object TaskSchedulerImpl {
 
     retval.toList
   }
+
+  private def maybeCreateBlacklistTracker(conf: SparkConf): Option[BlacklistTracker] = {
```

Comment:
the scope becomes a little wider, though -- its only used in `TaskSchedulerImpl`, and I'd have to make it `private[scheduler]`.  If it were there, I worry that it might look like a way to get a hold of the global singleton, like the `getOrCreate` methods on other objects.

I see it both ways, but think I feel a bit better about leaving it here so its `private`.  what do you think?